packages/common/src/pivot/derivePivotConfigFromChart.ts

```diff
@@ -147,63 +143,50 @@ function getCartesianPivotConfiguration(
         throw new Error('Invalid cartesian chart config - no eCharts config');
     }
 
-    const { eChartsConfig } = chartConfig.config;
-    const series = eChartsConfig?.series;
```

Comment:
We were deriving pivot information from the series, which was incorrect and caused a few bugs. Series are constantly being regenerated based on the results. 
We can infer the pivot configuration from the chart layout and the pivot columns.